ui/shared/components/panel/variants/VariantIndividuals.jsx

```diff
@@ -377,10 +363,166 @@ const Genotype = React.memo(({ variant, individual, isCompoundHet, genesById })
           {filters}
         </small>
       )}
+    </div>
+  )
+}
+
+GenotypeQuality.propTypes = {
+  genotype: PropTypes.object,
+  variant: PropTypes.object,
+}
+
+const LegacyAdditionalSampleTypePopup = ({ genotype, variant, isHemiX, genesById }) => {
+  const hasConflictingNumAlt = genotype.otherSample && genotype.otherSample.numAlt !== genotype.numAlt
+
+  return (
+    <div>
+      {genotype.otherSample && (
+      <Popup
+        header="Additional Sample Type"
+        trigger={<Icon name="plus circle" color={hasConflictingNumAlt ? 'red' : 'green'} />}
+        content={
+          <div>
+            {hasConflictingNumAlt && (
+              <div>
+                <VerticalSpacer height={5} />
+                <Alleles genotype={genotype.otherSample} variant={variant} isHemiX={isHemiX} />
+                <VerticalSpacer height={5} />
+              </div>
+            )}
+            {genotypeDetails(genotype.otherSample, variant, genesById)}
+          </div>
+        }
+      />
+      )}
+    </div>
+  )
+}
+
+LegacyAdditionalSampleTypePopup.propTypes = {
+  genotype: PropTypes.object,
+  variant: PropTypes.object,
+  isHemiX: PropTypes.bool,
+  genesById: PropTypes.object,
+}
+
+export const MultiSampleTypeAlleles = React.memo(({ genotypes, variant, individual, isHemiX, isCompoundHet }) => {
+  const warnings = genotypes.reduce((acc, genotype) => {
+    const warning = getWarningsForGenotype(genotype, variant, individual, isHemiX, isCompoundHet)
+    if (warning) {
+      acc[genotype.sampleType] = warning
+    }
+    return acc
+  }, {})
+
+  const hasDifferentNumAlt = genotypes.some(genotype => genotype.numAlt !== genotypes[0].numAlt)
+
+  return (
+    <Alleles
+      genotype={genotypes[0]}
+      variant={variant}
+      isHemiX={isHemiX}
+      warning={(Object.keys(warnings).length > 0 || hasDifferentNumAlt) && (
+        <div>
+          {Object.entries(warnings).map(([sampleType, warning]) => (
+            <div key={sampleType}>
+              <b>
+                {sampleType}
+                Warning:
+              </b>
+              {warning}
+            </div>
+          ))}
+          {hasDifferentNumAlt && (
+            <span>
+              <b>Warning: </b>
+              Genotypes differ across sample types
+              {genotypes.map(genotype => (
+                <div key={genotype.sampleType}>
+                  {genotype.sampleType}
+                  :
+                  <AlleleContainer>
```

Comment:
you can just use the `Alleles` class here, as without a warning prop or an SV (which can't have multiple sample types) this is just the same code as that class